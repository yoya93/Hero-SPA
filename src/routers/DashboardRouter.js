import React from "react";
import { Navbar } from "../components/ui/Navbar";
import { MarvelScreen } from "../components/marvel/MarvelScreen";
import { Route, Switch, Redirect } from "react-router-dom";
import { DcScreen } from "../components/dc/DcScreen";
import { HeroeScreen } from "../heroes/HeroeScreen";
import { Search } from "../components/search/Search";

export const DashboardRouter = () => {
  return (
    <>
      <Navbar />

      <div className="container mt-2">
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/hero/:heroeId" component={HeroeScreen} />
          <Route path="/dc" component={DcScreen} />
          <Route path="/search" component={Search} />

          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
