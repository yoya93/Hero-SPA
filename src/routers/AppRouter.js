import React, { useContext } from "react";

import { BrowserRouter as Router, Switch } from "react-router-dom";
import { LoginScreen } from "../components/login/LoginScreen";
import { DashboardRouter } from "./DashboardRouter";
import { PrivateRoute } from "./PrivateRoute";
import { AuthContext } from "../auth/AuthContext";
import { PublicRoute } from "./PublicRoute";

export const AppRouter = () => {
  const { user } = useContext(AuthContext);
  return (
    <Router basename="/heroes-app">
      <div>
        <Switch>
          <PublicRoute
            isAuthenticated={user.logged}
            exact
            path="/login"
            component={LoginScreen}
          />

          <PrivateRoute
            isAuthenticated={user.logged}
            path="/"
            component={DashboardRouter}
          />
        </Switch>
      </div>
    </Router>
  );
};
