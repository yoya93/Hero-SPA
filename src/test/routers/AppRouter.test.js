import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { AppRouter } from "../../routers/AppRouter";
import { AuthContext } from "../../auth/AuthContext";

describe("Pruebas en <AppRouter/>", () => {
  test("debe de mostrar el login si no está autenticado ", () => {
    const contextValue = {
      user: {
        logged: false,
      },
      dispatch: jest.fn(), // el dispatch es ua function y mediante jest puedo hacerl el tracking
    };

    // tenemos que darle el context que usa la app
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el navBar si  está autenticado ", () => {
    const contextValue = {
      user: {
        logged: true,
        name: "Yoya",
      },
      dispatch: jest.fn(), // el dispatch es ua function y mediante jest puedo hacerl el tracking
    };

    // tenemos que darle el context que usa la app
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find(".navbar").exists()).toBe(true);
  });
});
