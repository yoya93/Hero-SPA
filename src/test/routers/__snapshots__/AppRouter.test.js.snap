// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> debe de mostrar el login si no est√° autenticado  1`] = `
<div>
  <div
    className="container"
  >
    <h1>
      LoginScreen
    </h1>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
